api/service/projects_service.go

```diff
@@ -76,6 +81,25 @@ func (service *projectsService) CreateProject(ctx context.Context, project *mode
 		}
 	}
 
+	if service.webhookManager != nil {
+		err = service.webhookManager.InvokeWebhooks(ctx, ProjectCreatedEvent, project, func(p []byte) error {
+			// Expects webhook output to be a project object
+			var tmpproject models.Project
+			if err := json.Unmarshal(p, &tmpproject); err != nil {
+				return err
+			}
+			project, err = service.save(&tmpproject)
+			if err != nil {
+				return err
+			}
+			return nil
+		}, webhooks.NoOpErrorHandler)
+		if err != nil {
```

Comment:
I think this part probably the only real concern i have. Before reading this comment, do read the other comments first.

- If we have a single sync webhook that we can ignore on error,  it makes logically sense that it doesn't fail here (anyway the error is not being propagated)
- If I use an async webhook, this will never happen. 

Also we inherently meant that async webhook == results we dont care about, while we care about sync webhook, but ignoring error kinda describe it as an async webhook as well.

The concern here is, how we configure the webhooks have an impact on the client. Ie, the fact we handled error, we should never be using async webhook nor using sync webhook that allow failures (or we would need to knowingly handle it in the `onSuccess` method). 

Maybe this is unavoidable to the natural complexity and trying to abstract it for varying scenario. While we try to simplify some other areas (if possible), I think the existing readme is not sufficient to guide new dev on how exactly to implement the client with the desired webhooks in mind.

Or, can we delegate everything to a single webhook and let it own the logic and avoid the trap of falling into these misconfigurations?

Reply:
> The concern here is, how we configure the webhooks have an impact on the client. Ie, the fact we handled error, we should never be using async webhook nor using sync webhook that allow failures (or we would need to knowingly handle it in the onSuccess method).

Can you elaborate a bit more on this

Reply:
Ok do you mean, the way the webhook configuration is provided will affect how the client needs to process the response?

Reply:
The aim of this package is to allow different webhook configurations at the same time not needing to change the code based on the configuration provided. The final response expected and the input payload is up to the library user to define..So as long as the webhook configuration adheres to what the library callback and error handler expects there should be no need to change the existing logic. So the webhook configuration needs to match the code, not the other way round (if that makes sense)

Reply:
Perhaps to reduce complexity, the default should be to abort sync webhooks on failure, and only use the ignore failure as an escape hatch?

Reply:
> not needing to change the code based on the configuration provided.
currently, the client will fail on error anyway. 

> So the webhook configuration needs to match the code
yes but as seen, there needs to be a fair bit of understand in order to know how to handle or orchestrate the behaviour

Or at least ignore failure and having to handle that error in `onSuccess` is the most unintuitive to me about it. Or when mark it allowed to handle but having to handle it still.

Reply:
ok that makes sense. So it will be more straightforward to the caller that onSuccess, the payload passed in is always correct. Thanks for the clarification.